### PR TITLE
feat: add ruby support to dns-server challenge.

### PR DESCRIPTION
**Pull request by ryan-gang**

### CODE REVIEW

File: compiled_starters/ruby/app/server.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  #     response = ''
+  #     udp_socket.send(response, 0, source[3], source[1])
+  #   rescue => e
+  #     puts "Error receiving data: #{e}"
+  #     break
+  #   end
+  # end
+end
+
+start_server
```
Comment (rohitpaulk) [inline compiled_starters/ruby/app/server.rb:24]: Let's remove the function, just inline the code in the file directly

File: compiled_starters/ruby/app/server.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  # You can use print statements as follows for debugging, they'll be visible when running tests.
+  print("Logs from your program will appear here!")
+
+  # Uncomment this block to pass the first stage
+  #
+  # loop do
+  #   begin
+  #     buf, source = udp_socket.recvfrom(512)
+  #     response = ''
+  #     udp_socket.send(response, 0, source[3], source[1])
```
Comment (rohitpaulk) [inline compiled_starters/ruby/app/server.rb:16]: Can we use better variable names here? Isn't clear what `source[3]` and `source[1]` are.

File: compiled_starters/ruby/app/server.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  udp_socket.bind('127.0.0.1', 2053)
+
+  # You can use print statements as follows for debugging, they'll be visible when running tests.
+  print("Logs from your program will appear here!")
+
+  # Uncomment this block to pass the first stage
+  #
+  # loop do
+  #   begin
+  #     buf, source = udp_socket.recvfrom(512)
```
Comment (rohitpaulk) [inline compiled_starters/ruby/app/server.rb:14]: Can we rename `buf` to something like `data` or `message`? Seems more descriptive for this case. For example, https://ruby-doc.org/stdlib-1.9.1/libdoc/socket/rdoc/Socket.html#method-i-recvfrom uses `data` in their example

File: compiled_starters/ruby/app/server.rb (hunk 2 above)
Comment (rohitpaulk) [reply compiled_starters/ruby/app/server.rb:16]: Not sure if pattern matching in Ruby works this way, but maybe we could do something like: 

> message, [_, sender_port, _, sender_address] = socket.recvfrom(1024)

**Review by rohitpaulk — APPROVED** [review]

**Comment by ryan-gang** [comment]
@rohitpaulk fixed some `standardrb` warnings in server.rb, re-requesting.

**Review by rohitpaulk — APPROVED** [review]

**Pull request merged**